DHT/Rebalance - removing an unneeded subvol calculation and comparison

BarakSason opened the pull request:
During the data migration phase of rebalance, an unneeded calculation
and comparison is made. These operations are not required, as the
information is already cached.
Removing the unneeded operations.

fixes: ##1801
Change-Id: I9bce8152c7936a73edc4704b2f9d4f0f241ce13e
Signed-off-by: Barak Sason Rofman <bsasonro@redhat.com>

BarakSason committed:
DHT/Rebalance - removing an unneeded subvol calculation and comparison

During the data migration phase of rebalance, an unneeded calculation
and comparison is made. These operations are not required, as the
information is already cached.
Removing the unneeded operations.

fixes: ##1801
Change-Id: I9bce8152c7936a73edc4704b2f9d4f0f241ce13e
Signed-off-by: Barak Sason Rofman <bsasonro@redhat.com>

BarakSason commented:
Note - at a couple of places formatting was changed automatically.

BarakSason commented:
/run regression

BarakSason commented:
/recheck smoke

amarts commented on xlators/cluster/dht/src/dht-rebalance.c:
```diff
@@ -2322,17 +2312,32 @@ rebalance_task(void *data)
-                           local->rebalance.target_node, local->flags,
-                           &fop_errno);
 
+    hashed_subvol = local->rebalance.target_node;
```
is hashed_subvol same as 'conf->hashed_subvol'?

I vaguely remember the issue of target being different from hashed_subvol when disks are getting full on that node, but there is space in other nodes. Please check that. If that is confirmed that `rebalance.target_node` is always same as hashed_subvol, then this makes sense.

BarakSason replied:
@amarts There is no functional change here, just a removal of unneeded operations.

Look at the code prior to the changes - we pass local->rebalance.from_subvol and local->rebalance.target_node, so I just created local variables for these for readability.
In addition, in the code of `dht_migrate_file` we do:

```
if (from == to) {
        gf_msg_debug(this->name, 0,
                     "destination and source are same. file %s"
                     " might have migrated already",
                     loc->path);
        ret = 0;
        goto out;
    }
```
I moved this check out of `dht_migrate_file` to `rebalance_task`.

In addition, at lint 1594 we used to do a subvol calculation, but this is not needed as we already have this information from the calling method `gf_defrag_migrate_single_file`.

Hope this clear things out. Let me know if you have doubts.

amarts replied:
:+1:  Ok, good from my end then.

Looks like you need to do a resubmit as there is conflict. Will approve after that.

BarakSason replied:
Yes, I'll address this shortly.

csabahenk reviewed (dismissed):
- You have a ton of formatting-only edit, regarding which I don't even need to raise my subjective eyebrow that it seems just noise to me, because _clang-format_ formats them back to how they were before. The last stance I know in this regard is that _clang-format_ should output identical content as its input on our C sources, and that CI enforces this. This is obviously not up to date knowledge, as this patch has passed CI (I wonder how so); nevertheless, AFAIK this requirement is still in place, even if not enforced.
- You are making _dht_migrate_file_ more special, having more assumptions on the caller context (_from != to_ and _to_ being the hashed subvol).
   - Please make a note of these assumptions in the doc text.
   - Please consider making the function static, as thus it moves further from being suitable for general usage, and actually it ocurs in just a single file.

BarakSason committed:
DHT/Rebalance - removing an unneeded subvol calculation and comparison
(2)

During the data migration phase of rebalance, an unneeded calculation
and comparison is made. These operations are not required, as the
information is already cached.
Removing the unneeded operations.

fixes: ##1801
Change-Id: I9bce8152c7936a73edc4704b2f9d4f0f241ce13e
Signed-off-by: Barak Sason Rofman <bsasonro@redhat.com>

BarakSason commented:
>     * You have a ton of formatting-only edit, regarding which I don't even need to raise my subjective eyebrow that it seems just noise to me, because _clang-format_ formats them back to how they were before. The last stance I know in this regard is that _clang-format_ should output identical content as its input on our C sources, and that CI enforces this. This is obviously not up to date knowledge, as this patch has passed CI (I wonder how so); nevertheless, AFAIK this requirement is still in place, even if not enforced.
> 
>     * You are making _dht_migrate_file_ more special, having more assumptions on the caller context (_from != to_ and _to_ being the hashed subvol).
>       
>       * Please make a note of these assumptions in the doc text.
>       * Please consider making the function static, as thus it moves further from being suitable for general usage, and actually it ocurs in just a single file.

- Changed var names to match the convention in other places.
- Modified method scope to static

BarakSason commented:
/run regression
